pkg/client/metrics/metrics.go

```diff
@@ -23,25 +23,27 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
 	"sigs.k8s.io/prometheus-adapter/pkg/client"
 )
 
 var (
 	// queryLatency is the total latency of any query going through the
 	// various endpoints (query, range-query, series).  It includes some deserialization
 	// overhead and HTTP overhead.
-	queryLatency = prometheus.NewHistogramVec(
-		prometheus.HistogramOpts{
+	queryLatency = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
 			Name:    "cmgateway_prometheus_query_latency_seconds",
 			Help:    "Prometheus client query latency in seconds.  Broken down by target prometheus endpoint and target server",
-			Buckets: prometheus.ExponentialBuckets(0.0001, 2, 10),
+			Buckets: prometheus.ExponentialBuckets(0.001, 2, 10),
 		},
 		[]string{"endpoint", "server"},
 	)
 )
 
 func init() {
-	prometheus.MustRegister(queryLatency)
+	legacyregistry.MustRegister(queryLatency)
```

Comment:
We shouldn't add metrics to the Kubernetes global registry. We should rather dependency inject the Prometheus registry like it was done in the past in metrics-server: https://github.com/kubernetes-sigs/metrics-server/pull/587/. There was a few changes made to the original work in https://github.com/kubernetes-sigs/metrics-server/pull/634/, but I think we should follow the same process in prometheus-adapter.